x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/rule_diff_tab_react_diff_view.tsx

```diff
@@ -0,0 +1,598 @@
+  const hunksWithMinLinesCollapsed = useMinCollapsedLines(0, hunksWithSourceExpanded, oldSource);
+
+  return {
+    expandRange,
+    hunks: hunksWithMinLinesCollapsed,
+  };
+};
+
+const useTokens = (hunks: HunkData[], compareMethod: DiffMethod, oldSource: string) => {
+  if (!hunks) {
+    return undefined;
+  }
+
+  const options: TokenizeOptions = {
+    oldSource,
+    highlight: false,
+    enhancers: [
+      /*
+        "markEditsBy" is a slightly modified version of "markEdits" enhancer from react-diff-view 
+        to enable word-level highlighting.
+      */
+      compareMethod === DiffMethod.CHARS
+        ? markEdits(hunks, { type: 'block' }) // Using built-in "markEdits" enhancer for char-level diffing
+        : markEditsBy(hunks, compareMethod), // Using custom "markEditsBy" enhancer for other-level diffing
+    ],
+  };
+
+  try {
+    /*
+      Synchroniously applies all the enhancers to the hunks and returns an array of tokens.
+      There's also a way to use a web worker to tokenize in a separate thread.
+      Example can be found here: https://github.com/otakustay/react-diff-view/blob/49cebd0958ef323c830395c1a1da601560a71781/site/components/DiffView/index.tsx#L43
+      It didn't work for me right away, but theoretically the possibility is there.
+    */
+    return tokenize(hunks, options);
+  } catch (ex) {
+    return undefined;
+  }
+};
+
+const convertToDiffFile = (oldSource: string, newSource: string) => {
+  /* 
+    "diffLines" call below converts two strings of text into an array of Change objects.
+    Change objects look like this:
+    [
+      ...
+      {
+        "count": 2,
+        "removed": true,
+        "value": "\"from\": \"now-540s\""
+      },
+      {
+        "count": 1,
+        "added": true,
+        "value": "\"from\": \"now-9m\""
+      },
+      ...
+    ]
+
+    "formatLines" takes an array of Change objects and turns it into one big "unified Git diff" string.
+    Unified Git diff is a string with Git markers added. Looks something like this:
+    `
+      @@ -3,16 +3,15 @@
+         "author": ["Elastic"],
+      -  "from": "now-540s",
+      +  "from": "now-9m",
+         "history_window_start": "now-14d",
+    `
+  */
+
+  const unifiedDiff: string = unidiff.formatLines(unidiff.diffLines(oldSource, newSource), {
+    context: 3,
+  });
+
+  /*
+      "parseDiff" converts a unified diff string into a JSDiff File object.
+    */
+  const [diffFile] = parseDiff(unifiedDiff, {
+    nearbySequences: 'zip',
+  });
+  /*
+      File object contains some metadata and the "hunks" property - an array of Hunk objects.
+      At this stage Hunks represent changed lines of code plus a few unchanged lines above and below for context.
+      Hunk objects look like this:
+      [
+        ...
+        {
+          content: '  "from": "now-9m",'
+          isInsert: true,
+          lineNumber: 14,
+          type: "insert"
```

Comment:
I think these 3 properties are in the live within the `changes` object, which is a sibling of `content`?